src/components/Table/TableSaveViewModal/TableSaveViewModal.test.jsx

```diff
@@ -0,0 +1,260 @@
+
+describe('TableSaveViewModal', () => {
+  const { i18n } = TableSaveViewModal.defaultProps;
+  const actions = {
+    onSave: jest.fn(),
+    onClose: jest.fn(),
+    onClearError: jest.fn(),
+    onChange: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('passes along all input data onSave callback', () => {
+    const { getByText, getByTestId, getByLabelText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" />
+    );
+    fireEvent.change(getByTestId('my-modal-form-title-input'), { target: { value: 'testval1' } });
+    fireEvent.click(getByText(i18n.saveButtonLabelText));
+
+    expect(actions.onSave).toHaveBeenCalledWith({
+      isDefault: false,
+      isPublic: false,
+      title: 'testval1',
+    });
+
+    fireEvent.change(getByTestId('my-modal-form-title-input'), { target: { value: 'testval2' } });
+    fireEvent.click(getByLabelText(i18n.defaultCheckboxLabelText));
+    fireEvent.click(getByLabelText(i18n.publicCheckboxLabelText));
+    fireEvent.click(getByText(i18n.saveButtonLabelText));
+
+    expect(actions.onSave).toHaveBeenCalledWith({
+      isDefault: true,
+      isPublic: true,
+      title: 'testval2',
+    });
+  });
+
+  it('calls onChange when form elements value changes', () => {
+    const { getByTestId, getByLabelText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" />
+    );
+    fireEvent.change(getByTestId('my-modal-form-title-input'), { target: { value: 'testval1' } });
+    fireEvent.click(getByLabelText(i18n.defaultCheckboxLabelText));
+    fireEvent.click(getByLabelText(i18n.publicCheckboxLabelText));
+
+    expect(actions.onChange).toHaveBeenCalledWith({
+      title: 'testval1',
+    });
+    expect(actions.onChange).toHaveBeenCalledWith({
+      isDefault: true,
+    });
+    expect(actions.onChange).toHaveBeenCalledWith({
+      isPublic: true,
+    });
+  });
+
+  it('calls onClose on cancel and close-icon click', () => {
+    const { getByRole, getByText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" />
+    );
+    fireEvent.click(getByText(i18n.cancelButtonLabelText));
+    fireEvent.click(getByRole('button', { name: i18n.closeIconDescription }));
+
+    expect(actions.onClose).toHaveBeenCalledTimes(2);
+  });
+
+  it('handles errors correctly', () => {
+    const { getByRole, queryByText } = render(
+      <TableSaveViewModal actions={actions} open testID="my-modal" error="my test error" />
+    );
+    expect(queryByText('my test error')).toBeInTheDocument();
+    expect(actions.onClearError).not.toHaveBeenCalled();
+
+    fireEvent.click(getByRole('button', { name: 'closes notification' }));
+    expect(queryByText('my test error')).not.toBeInTheDocument();
+    expect(actions.onClearError).toHaveBeenCalled();
+  });
+
+  it('replaces the strings with i18n strings', () => {
+    const i18nTest = {
+      modalTitle: 'modal-title-test',
```

Comment:
I think like this idea. We've not done this before, but makes sense to ensure that i18n strings are appropriately used when configured to prevent silent regressions moving forward. In theory prop types and default props should ensure these are always used, but the variable could be used without actually being rendered to the DOM. 

In the future we might be able to achieve this easier or automatically via snapshots of some kind.

Reply:
Cool, having this automated would be great.